src/components/Sidebars/configs.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from "react";
+import { VscJson } from "react-icons/vsc";
+import { ConfigItem, getConfigsBy } from "../../api/services/configs";
+import CollapsiblePanel from "../CollapsiblePanel";
+import ConfigLink from "../ConfigLink/ConfigLink";
+import EmptyState from "../EmptyState";
+import { Loading } from "../Loading";
+import Title from "../Title/title";
+
+type Props = {
+  topologyId?: string;
+  configId?: string;
+};
+
+export function ConfigsList({ topologyId, configId }: Props) {
+  const [configs, setConfigs] = useState<ConfigItem[]>([]);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchConfigs() {
```

Comment:
Just something I have been meaning to bring up, we probably should start of switching all API calls to React Query; the User Experience will greatly improve due to its fine-grained state management and caching, making the data appear much faster.